### DISCUSSION TITLE
Cannot clone Github repo to Raven

### DISCUSSION THREAD

**Discussion by sr320** [RobertsLab/resources#1332] [Q&A]
Trying

<img width="530" alt="Screen Shot on 2021-11-01 at 12-41-46" src="https://user-images.githubusercontent.com/4442039/139732516-75f81a23-2524-49f9-a412-7c0211995010.png">

get 

```
>> /usr/bin/git clone --progress https://github.com/epigeneticstoocean/2018_L18-adult-methylation.git 2018_L18-adult-methylation
Cloning into '2018_L18-adult-methylation'...
error: chmod on /media/8TB_ext_HDD01/sr320/github/2018_L18-adult-methylation/.git/config.lock failed: Operation not permitted
fatal: could not set 'core.filemode' to 'false'

```

**Comment by kubu4** [comment]
Oh boy...

Annoying workaround would be to clone repo to one of the other HDDs and then move the entire repo to the 8TB HDD.

The "real" fix is to reformat that drive and change it to a different file system that is compatible with Unix permissions...

**Reply by sr320** [reply]
Tried that originally - also failed. 

rsync failed

**Reply by kubu4** [reply]
Wait both `mv` command _and_ `rsync` failed?

What was your `rsync` command?

**Reply by sr320** [reply]
did not try `mv` just `rsync`... will try `mv` now...

`rsync -avz`

**Reply by sr320** [reply]
actually will try `cp`

**Reply by kubu4** [reply]
Ah! For the `rsync` command, you won't be able to use the `-a` option with that drive... Another annoying "nuance" with the file system set for that HDD...

**Reply by sr320** [reply]
rsync without the `-a` does not move anything... trying `cp -r` now.

**Reply by sr320** [reply]
re `rsync`

```
sr320@raven:/media/8TB_ext_HDD01/sr320/github$ rsync -vz /home/sr320/github/2018_L18-adult-methylation/ 2018_L18-adult-methylation/
skipping directory .

sent 16 bytes  received 12 bytes  56.00 bytes/sec
total size is 0  speedup is 0.00
sr320@raven:/media/8TB_ext_HDD01/sr320/github$ rsync -vz /home/sr320/github/2018_L18-adult-methylation/* 2018_L18-adult-methylation/
skipping directory analyses
skipping directory bg_data
skipping directory code
skipping directory data
2018_L18-adult-methylation.Rproj
README.md

sent 469 bytes  received 54 bytes  1,046.00 bytes/sec
total size is 383  speedup is 0.73
sr320@raven:/media/8TB_ext_HDD01/sr320/github$ rsync -vz /home/sr320/github/2018_L18-adult-methylation .
skipping directory 2018_L18-adult-methylation

sent 16 bytes  received 12 bytes  56.00 bytes/sec
total size is 0  speedup is 0.00
```

**Reply by kubu4** [reply]
I think you need to add `-r` to put it into recursive mode (which is part of what the `-a` option does).

**Comment by sr320** [comment]
Stepping back for a minute- the HD is not a solution to the problem of no space on raven.

I use raven for Rstudio / Github... I need to be able to work in external HD

**Reply by kubu4** [reply]
I don't think I follow...

**Reply by sr320** [reply]
Reason for the external hard drives is to free up space on raven proper.
I cannot work on external hard drive because drive format does not let me clone github repos to...

**Reply by sr320** [reply]
I suppose I could maybe start new repo on external HD? then work (commit/pull).. (cannot test right now as I am using Rstudio ATM)

But presume pull would not work.

**Comment by kubu4** [comment]
The 8TB external HDD attached to my computer is formatted in a Linux-friendly format (ext4). This means that permission changes and timestamp changes can be manipulated via traditional commands (e.g. `chmod`, `chown`, `chgrp`, etc) which would probably simplify our usage if we swapped this for the one currently attached to Raven.

If you'd like to do this, I've already "unmounted" the HDD attached to my computer, so you can just pull it out of the toaster without worry. Along those lines, we should unmount the 8TB external attached to Raven prior to removal. I'm guessing you could do this via the GUI. Otherwise, let me know and I can unmount it remotely.

**Reply by sr320** [reply]
GUI is unresponsive - if you want to unmount.

**Reply by kubu4** [reply]
I've unmounted the 8TB HDD on Raven.

**Reply by sr320** [reply]
switched...
Note I do not have write permissions.
```
sr320@raven:/media/8TB_ext_HDD01$ mkdir sr320
mkdir: cannot create directory ‘sr320’: Permission denied
```

**Reply by kubu4** [reply]
I'll get this taken care of in a couple of minutes, but just wanted to add some info.

> Note I do not have write permissions.

I think this is to be expected (at the moment). Since we've disconnected the old drive and added a new drive, that directory exists in its original state when it was created. Meaning, directories in `/media` default to having `root` as owner and group. So, you'd have to use `sudo` to make any changes.

Secondly, the computer doesn't know where to mount the new drive, so we would not expect the new drive to get mounted in the same location as the old drive.

Thirdly, it's likely the new drive hasn't been mounted at all yet, so we'll need to tell the computer to both mount the drive, as well as the location we'd like it mounted.

Anyway, I'll report back once I've configured it to mount in that location. BRB...

**Reply by kubu4** [reply]
Ha! Well, that went just as smoothly as it's been going for the past week. Rebooted and now can't get in (again)...

**Reply by kubu4** [reply]
Believe it or not, the 8TB external HDD on Raven is up and running!!!!!! There shouldn't be any permission issues or rsync issues!

Please test and report back!

**Reply by sr320** [reply]
No permissions issue... (unless you try to clone a repo - https://d.pr/i/hosUFw)

**Reply by kubu4** [reply]
I'm in the same boat as well. Can't make a directory (again)...

**Reply by kubu4** [reply]
I have unmounted the 8TB HDD and am currently running `sudo fsck.ext4 -fy` on that drive. This looks for and attempts to fix any errors. It has found _numerous_ errors (crazy stuff about inodes, "imagic flag", timestamps...) and is supposedly fixing them. We'll see...

**Comment by kubu4** [comment]
Hallelujah! I think I've done it! Seriously. I moved the 8TB HDD to an internal drive bay. I've even upped the game by adding a _second_ 8TB HDD and an additional internal 1TB HDD!

Here're the details (I'll get these added to the Handbook shortly).

List of all HDDs and the locations (this list does _not_ include the primary 1TB boot drive):

```shell
/dev/sde1       916G  407G  464G  47% /media/1TB_ext_SSD01
/dev/sdf1       916G   77M  870G   1% /media/1TB_ext_SSD02
/dev/sdc1       7.3T  519M  6.9T   1% /home/shared/8TB_HDD_01
/dev/sdd1       7.3T   93M  6.9T   1% /home/shared/8TB_HDD_02
/dev/sdb1       916G   77M  870G   1% /home/shared/1TB_HDD02
```

I've briefly tested by successfully cloning the `RobertsLab/resources` repo to the three internal HDDs _and_ the computer didn't lock up!

Please try things out and let me know how it goes...

:pray:
